### ISSUE TITLE
getting "Cannot read property 'visualization' of undefined" after upgrade to grafana 8.0

### ISSUE THREAD

**Issue by vipin5050** [marcusolsson/grafana-gantt-panel#38]
Not sure if it related to upgrade but I noticed after upgrading to grafana 8.0 I am getting "Cannot read property 'visualization' of undefined, grafana"

<img width="405" alt="Screen Shot 2021-06-13 at 9 47 47 AM" src="https://user-images.githubusercontent.com/79691636/121815665-73e5cf00-cc2c-11eb-8bf7-ebbfdf4a67e0.png">

**Comment by marcusolsson** [comment]
Hi, and thanks for reporting this. I'm unfortunately not able to reproduce this. The plugin also doesn't have any property called _visualization_, which leads me to be believe this might be a Grafana bug. Does this happen for only the Gantt panel?

**Comment by marcusolsson** [comment]
Is there any way you could give me a (anonymized) dashboard JSON that reproduces this issue?

**Comment by vipin5050** [comment]
Hi @marcusolsson yes this happened only with gantt chart and after I saw this error with grafana 8.0.1, I downgraded to grafana 7.8.1 and it worked without any issue.

**Comment by vipin5050** [comment]
> Is there any way you could give me a (anonymized) dashboard JSON that reproduces this issue?

let me know where I can send you JSON

**Comment by marcusolsson** [comment]
Could you create a [Gist](https://gist.github.com/) maybe?

**Comment by vipin5050** [comment]
https://gist.github.com/vipin5050/43d4b38301616468b876afc542658f69#file-gistfile1-txt

let me know if you can see it

**Comment by marcusolsson** [comment]
Thank you, I'll take a look at it! Also, could you open the Developer Tools in your browser and check the Console tab and see if there's a stack trace with more information?

I've published a new version (0.7.1) with some minor fixes. I don't think there's anything in there that would affect this, but could you check the new version to confirm?

**Comment by trophi20** [comment]
I also got the error (compare [https://github.com/marcusolsson/grafana-gantt-panel/discussions/41](https://github.com/marcusolsson/grafana-gantt-panel/discussions/41))

Here is the log from the browser:
```
TypeError: t is undefined
    getCalculator fieldColor.ts:38
    x scale.ts:37
    S displayProcessor.ts:104
    GanttChart GanttChart.tsx:321
    GanttChart GanttChart.tsx:299
    React 7
        na
        Uu
        Os
        _s
        ws
        ds
        qo
    unstable_runWithPriority scheduler.production.min.js:18
    React 13
        Ho
        qo
        Wo
        Re
        $t
        jr
        Ar
        Cr
        Cr
        Xs
        _reactRootContainer
        Qs
        render
    init app.ts:111
    promise callback*init app.ts:110
    zl43 index.ts:20
    Webpack 4
        o
        r
        t
        <anonymous>
react-dom.production.min.js:216:191
    React 5
        iu
        callback
        li
        du
        Ss
    unstable_runWithPriority scheduler.production.min.js:18
    React 4
        Ho
        Es
        ds
        qo
    unstable_runWithPriority scheduler.production.min.js:18
    React 13
        Ho
        qo
        Wo
        Re
        $t
        jr
        Ar
        Cr
        Cr
        Xs
        _reactRootContainer
        Qs
        render
    init app.ts:111
    (Async: promise callback)
    init app.ts:110
    zl43 index.ts:20
    Webpack 4
        o
        r
        t
        <anonym>

```

**Comment by marcusolsson** [comment]
I've been testing this on Grafana 8.0.0, 8.0.1, and 8.0.2, with the latest version of the plugin (0.7.1), on a macOS 11.2.3, on Firefox, Safari, and Chrome, and fail to replicate this issue. Could you perhaps share your environment details?

Is there any specific configuration that produces the error, or does the error occur with default settings?

**Comment by trophi20** [comment]
I'm working on 
- windows 10
- grafana 8.0.1
- gantt plugin 0.7.1
- firefox 89.0

Do you need some more information? 

By the way: I tried it with grafana version 7.5.7 With this version it worked.

**Comment by vipin5050** [comment]
> I'm working on
> 
> * windows 10
> * grafana 8.0.1
> * gantt plugin 0.7.1
> * firefox 89.0
> 
> Do you need some more information? (Query, etc)
> 
> By the way: I tried it with grafana version 7.5.7 With this version it worked.

Hi @trophi20 so it worked with grafana 7.5.7 but does not work with grafana 8.0.1? same here same setup worked with grafana 7.5.7 but does not work with 8.0.1

**Comment by marcusolsson** [comment]
I think I found the problem. Could you try 0.7.2 and let me know if the issue's still there?

**Comment by vipin5050** [comment]
Hi @marcusolsson I tried again with 0.7.2 and grafana 8.0.1, issue is still there. the moment I downgrade grafana to 7.5.1 it worked.

**Comment by trophi20** [comment]
I took the 0.7.2 from github and builded it. I used grafana 8.0.1, but it is still not working:
`TypeError: t is undefined
    getCalculator fieldColor.ts:38
    x scale.ts:37
    S displayProcessor.ts:104
    GanttChart GanttChart.tsx:321
    GanttChart GanttChart.tsx:299
    React 7
        na
        Uu
        Os
        _s
        ws
        ds
        qo
    unstable_runWithPriority scheduler.production.min.js:18
    React 6
        Ho
        qo
        Wo
        us
        enqueueSetState
        setState
    onDataUpdate PanelChrome.tsx:221
    next PanelChrome.tsx:127
    RxJS 19
        __tryOrUnsub
        next
        _next
        next
        _next
        next
        notifyNext
        _next
        next
        r
        _trySubscribe
        subscribe
        c
        _innerSub
        _tryNext
        _next
        next
        next
        nextInfiniteTimeWindow
    next PanelQueryRunner.ts:268
    RxJS 48
        __tryOrUnsub
        next
        _next
        next
        notifyNext
        _next
        next
        _next
        next
        next
        _next
        next
        _next
        next
        _next
        next
        _next
        next
        _next
        next
        _next
        next
        _next
        next
        _next
        next
        _next
        next
        _next
        next
        _next
        next
        notifyNext
        _next
        next
        _next
        next
        _next
        next
        _next
        next
        next
        _next
        next
        notifyNext
        _next
        next
        l
react-dom.production.min.js:216:191
`

**Comment by vipin5050** [comment]
@marcusolsson any update, do you need any more information?

**Comment by marcusolsson** [comment]
I'm a bit stumped here tbh. I know what the problem is—the theme is not being set—but I can't seem to find where 😅  Also I'm still not able to reproduce this so it's hard to debug. The only difference between our setups is that you're both running on Windows 10. I'll try to find a Windows machine to test this on next week.

**Comment by vipin5050** [comment]
> I'm a bit stumped here tbh. I know what the problem is—the theme is not being set—but I can't seem to find where 😅 Also I'm still not able to reproduce this so it's hard to debug. The only difference between our setups is that you're both running on Windows 10. I'll try to find a Windows machine to test this on next week.

I am running on linux, if you want I can work online witn you and show the issue firsthand. could it be because starting grafana 8 you have to explicitly allow plugin to load if its not signed, just a guess.

**Comment by jeje8365** [comment]
Hi ! 
I have the same problem on my Grafana. I put my data here if it can help you.
 
Version of Grafana : 8.0.2
Version of plugin : 0.7.2
Datasource : Elasticsearch
Query : See below
Transform : Filter by name to remove "Count"
 
![image](https://user-images.githubusercontent.com/86005394/122737662-922d7980-d281-11eb-8563-5a3b5a527453.png)

I see different error depending on the setup: 
    * On Windows 10 + Firefox 78.11.0esr      "t is undifined"
    * On same Windows 10 + Edge 83.0.478.56        "cannot read property 'visualization' of undefined"
    *  On other computer, Ubuntu 18.04 + Firefox 89.0          "t is undifined"

If it helps, I can provide the log for each browser.

Interessant point : The same query+transform used on your "Calendar" extension and it works fine. 
Sadly, I migrated directly from Grafana 7.0 to 8.0, so I can't tell if the plugin worked with a 7.3+ with my setup... In fact, I migrated to Grafana 8.0 to be able to use this plugin :-) !

I hope it will help. 

Many thanks for all your work !

**Comment by benganellison** [comment]
I have the same issue, running mac as client, linux as host, elasticsearch as data source.  Works in 7.3 but not in 8.0

**Comment by marcusolsson** [comment]
Hope to get a chance to look at this during the weekend. Will set up a dual boot PC so that I can test on other configurations. Thanks for your patience! 🙏

**Comment by vipin5050** [comment]
@marcusolsson I am eagerly waiting for fix :-), let me know if I can help in testing on weekend, I in PST timezone.

**Comment by vipin5050** [comment]
Hi @marcusolsson did you get a chance to look in to the issue over the weekend?

**Comment by TatianaInfoGuild** [comment]
@marcusolsson I also have the same issue. Looking forward for the fix.

**Comment by TatianaInfoGuild** [comment]
Jfyi
I added only one gantt panel and see these errors:
![image](https://user-images.githubusercontent.com/40390186/125923964-c6f30ed8-ce6d-4283-96ae-aef0a227bdac.png)
The first two errors also present with other panels and they work.
I noticed the same error about 'visualization' with another panel: Waterfall. However, after pressing F5 it works ok, but Gantt keeps showing the same error.

**Comment by vipin5050** [comment]
@marcusolsson any update on the fix, will appreciate if could take a look in fixing this issue, our upgrade of grafana is stuck.

**Comment by marcusolsson** [comment]
Sorry for the lack of updates on this, I've been on vacation for the past few weeks. I'll start looking into this tomorrow!

**Comment by marcusolsson** [comment]
While I still haven't been able to reproduce this myself, I think I know what the issue is. I've published v0.7.3 with (hopefully) a fix. Could you try it out and let me know if that fixes the issue for you?

**Comment by vipin5050** [comment]
Thanks @marcusolsson its working, after upgrading to latest release issue is no more and I upgraded grafana to 8.1.0 too. Thanks for your support. BTW what was the issue?

**Comment by TatianaInfoGuild** [comment]
@marcusolsson thank you for the update! I will check and inform you later. Currently, I see that the behavior is different, however, seems to have some problems with my own query. I will get back to you about the final results.

**Comment by marcusolsson** [comment]
> BTW what was the issue?

I fixed the issue in a support library by adding an optional (for backwards-compatibility) argument to a helper that converts the start and end to time fields if they are string or number fields. I updated the dependency, but forgot to actually pass a value to the added argument. It didn't complain because it was optional.

I guess I just needed some time off to spot it 😅
